# Análisis y visualización de datos.
### Actividad Nro. 2

#### 1 - Librerias a utilizar

In [3]:
import pandas as pd
import numpy as np
import os

#### 2 - Carga de Bases de Datos

In [5]:
poblacion = pd.read_csv('poblacion.csv', encoding='latin-1')
esperanza = pd.read_csv('esperanza_de_vida.csv', encoding='latin-1')
hogares = pd.read_csv('hogares_viviendas_superficie.csv', encoding='latin-1')

#### 3 - Revisión de los datos

##### 3.1 - Muestra las primeras filas de cada DataFrame

In [ ]:
poblacion.head() 

provincia_id        provincia  hogares  viviendas_particulares  \
0             2  Capital Federal  1150134                 1423973   
1             6     Buenos Aires  4789484                 5377786   
2            10        Catamarca    96001                  113634   
3            14          Córdoba  1031843                 1232211   
4            18       Corrientes   267797                  292644   

   viviendas_particulares_habitadas  superficie_km2  
0                           1082998             200  
1                           4425193          307571  
2                             89376          102602  
3                            978553          165321  
4                            248844           88199

In [9]:
esperanza.head()

provincia  anio  mujeres  varones
0  Buenos Aires  2015    80.22    73.54
1  Buenos Aires  2020    81.34    74.74
2  Buenos Aires  2025    82.32    75.80
3  Buenos Aires  2030    83.20    76.76
4  Buenos Aires  2035    83.98    77.60

In [10]:
hogares.head()

provincia_id        provincia  hogares  viviendas_particulares  \
0             2  Capital Federal  1150134                 1423973   
1             6     Buenos Aires  4789484                 5377786   
2            10        Catamarca    96001                  113634   
3            14          Córdoba  1031843                 1232211   
4            18       Corrientes   267797                  292644   

   viviendas_particulares_habitadas  superficie_km2  
0                           1082998             200  
1                           4425193          307571  
2                             89376          102602  
3                            978553          165321  
4                            248844           88199

##### 3.2 - Muestra las ultimas filas de cada DataFrame

In [11]:
poblacion.tail()

provincia  anio  poblacion_total  poblacion_varones  \
770  Tierra del Fuego  2036           241593             122567   
771  Tierra del Fuego  2037           245734             124625   
772  Tierra del Fuego  2038           249853             126670   
773  Tierra del Fuego  2039           253948             128702   
774  Tierra del Fuego  2040           258020             130721   

     poblacion_mujeres  
770             119026  
771             121109  
772             123183  
773             125246  
774             127299

In [12]:
esperanza.tail()

provincia  anio  mujeres  varones
139   Tucumán  2020    81.05    75.11
140   Tucumán  2025    82.11    76.15
141   Tucumán  2030    83.03    77.07
142   Tucumán  2035    83.84    77.88
143   Tucumán  2040    84.54    78.58

In [13]:
hogares.tail()

provincia_id            provincia  hogares  viviendas_particulares  \
19            78           Santa Cruz    81796                   93881   
20            82             Santa Fe  1023777                 1143651   
21            86  Santiago del Estero   218025                  242034   
22            90              Tucumán   368538                  396040   
23            94     Tierra del Fuego    38956                   43360   

    viviendas_particulares_habitadas  superficie_km2  
19                             76233          243943  
20                            948369          133007  
21                            197906          136351  
22                            335821           22524  
23                             36689         1002445

##### 3.3 - Muestra la composición (filas y columnas) de cada DataFrame

In [18]:
print(poblacion.shape)
print(esperanza.shape)
print(hogares.shape)

(775, 5)
(144, 4)
(24, 6)


##### 3.4 - Muestra los tipos de datos de cada DataFrame

In [19]:
poblacion.dtypes

provincia            object
anio                  int64
poblacion_total       int64
poblacion_varones     int64
poblacion_mujeres     int64
dtype: object

In [20]:
esperanza.dtypes

provincia     object
anio           int64
mujeres      float64
varones      float64
dtype: object

In [22]:
hogares.dtypes

provincia_id                         int64
provincia                           object
hogares                              int64
viviendas_particulares               int64
viviendas_particulares_habitadas     int64
superficie_km2                       int64
dtype: object

##### 3.5 - Muestra un resumen estadistico de cada DataFrame

In [23]:
poblacion.describe()

anio  poblacion_total  poblacion_varones  poblacion_mujeres
count   775.000000     7.750000e+02       7.750000e+02       7.750000e+02
mean   2025.000000     3.777746e+06       1.856888e+06       1.920858e+06
std       8.950048     9.560571e+06       4.699604e+06       4.861043e+06
min    2010.000000     1.316610e+05       6.723500e+04       6.442600e+04
25%    2017.000000     5.845510e+05       2.906740e+05       2.934905e+05
50%    2025.000000     1.017731e+06       5.061010e+05       5.161370e+05
75%    2033.000000     1.855285e+06       9.138865e+05       9.404745e+05
max    2040.000000     5.277848e+07       2.603809e+07       2.674038e+07

In [24]:
esperanza.describe()

anio     mujeres     varones
count   144.000000  144.000000  144.000000
mean   2027.500000   82.710347   76.417778
std       8.568931    1.637181    1.749619
min    2015.000000   78.320000   71.530000
25%    2020.000000   81.525000   75.177500
50%    2027.500000   82.865000   76.595000
75%    2035.000000   84.117500   77.890000
max    2040.000000   85.460000   79.270000

In [25]:
hogares.describe()

provincia_id       hogares  viviendas_particulares  \
count     24.000000  2.400000e+01            2.400000e+01   
mean      48.000000  5.071531e+05            5.755052e+05   
std       28.284271  9.631395e+05            1.087472e+06   
min        2.000000  3.895600e+04            4.336000e+04   
25%       25.000000  1.369578e+05            1.513558e+05   
50%       48.000000  2.086070e+05            2.393215e+05   
75%       71.000000  3.701838e+05            4.034278e+05   
max       94.000000  4.789484e+06            5.377786e+06   

       viviendas_particulares_habitadas  superficie_km2  
count                      2.400000e+01    2.400000e+01  
mean                       4.715628e+05    1.567198e+05  
std                        8.908855e+05    1.940901e+05  
min                        3.668900e+04    2.000000e+02  
25%                        1.270420e+05    7.827275e+04  
50%                        1.942515e+05    1.011175e+05  
75%                        3.411782e+05    1.579462e+05  
max                        4.425193e+06    1.002445e+06

##### 3.6 - Aplicamos filtrado en DataFrame 'población' para excluir 'Total país'

In [30]:
poblacion.groupby('provincia')['provincia'].count()


provincia
Buenos Aires           31
Capital Federal        31
Catamarca              31
Chaco                  31
Chubut                 31
Corrientes             31
Córdoba                31
Entre Ríos             31
Formosa                31
Jujuy                  31
La Pampa               31
La Rioja               31
Mendoza                31
Misiones               31
Neuquén                31
Río Negro              31
Salta                  31
San Juan               31
San Luis               31
Santa Cruz             31
Santa Fe               31
Santiago del Estero    31
Tierra del Fuego       31
Total País             31
Tucumán                31
Name: provincia, dtype: int64

In [34]:
poblacion_filtrado = poblacion[poblacion['provincia'] != 'Total País']
poblacion_filtrado

provincia  anio  poblacion_total  poblacion_varones  \
31    Capital Federal  2010          3028481            1405566   
32    Capital Federal  2011          3033639            1409835   
33    Capital Federal  2012          3038860            1414105   
34    Capital Federal  2013          3044076            1418339   
35    Capital Federal  2014          3049229            1422507   
..                ...   ...              ...                ...   
770  Tierra del Fuego  2036           241593             122567   
771  Tierra del Fuego  2037           245734             124625   
772  Tierra del Fuego  2038           249853             126670   
773  Tierra del Fuego  2039           253948             128702   
774  Tierra del Fuego  2040           258020             130721   

     poblacion_mujeres  
31             1622915  
32             1623804  
33             1624755  
34             1625737  
35             1626722  
..                 ...  
770             119026  
771             121109  
772             123183  
773             125246  
774             127299  

[744 rows x 5 columns]

##### 3.7 - Creación un nuevo DataFrame por medio de la union de población_filtrado y hogares

In [35]:
poblac_superficie = pd.merge(poblacion_filtrado, hogares, left_on='provincia', right_on='provincia', how='left')
poblac_superficie

provincia  anio  poblacion_total  poblacion_varones  \
0     Capital Federal  2010          3028481            1405566   
1     Capital Federal  2011          3033639            1409835   
2     Capital Federal  2012          3038860            1414105   
3     Capital Federal  2013          3044076            1418339   
4     Capital Federal  2014          3049229            1422507   
..                ...   ...              ...                ...   
739  Tierra del Fuego  2036           241593             122567   
740  Tierra del Fuego  2037           245734             124625   
741  Tierra del Fuego  2038           249853             126670   
742  Tierra del Fuego  2039           253948             128702   
743  Tierra del Fuego  2040           258020             130721   

     poblacion_mujeres  provincia_id  hogares  viviendas_particulares  \
0              1622915             2  1150134                 1423973   
1              1623804             2  1150134                 1423973   
2              1624755             2  1150134                 1423973   
3              1625737             2  1150134                 1423973   
4              1626722             2  1150134                 1423973   
..                 ...           ...      ...                     ...   
739             119026            94    38956                   43360   
740             121109            94    38956                   43360   
741             123183            94    38956                   43360   
742             125246            94    38956                   43360   
743             127299            94    38956                   43360   

     viviendas_particulares_habitadas  superficie_km2  
0                             1082998             200  
1                             1082998             200  
2                             1082998             200  
3                             1082998             200  
4                             1082998             200  
..                                ...             ...  
739                             36689         1002445  
740                             36689         1002445  
741                             36689         1002445  
742                             36689         1002445  
743                             36689         1002445  

[744 rows x 10 columns]

##### 3.8 - Calculo de la densidad y se agrega resultado al DataFrame (población total / superficie)

In [37]:
poblac_superficie['densidad_poblacion'] = poblac_superficie['poblacion_total'] / poblac_superficie['superficie_km2']
poblac_superficie

provincia  anio  poblacion_total  poblacion_varones  \
0     Capital Federal  2010          3028481            1405566   
1     Capital Federal  2011          3033639            1409835   
2     Capital Federal  2012          3038860            1414105   
3     Capital Federal  2013          3044076            1418339   
4     Capital Federal  2014          3049229            1422507   
..                ...   ...              ...                ...   
739  Tierra del Fuego  2036           241593             122567   
740  Tierra del Fuego  2037           245734             124625   
741  Tierra del Fuego  2038           249853             126670   
742  Tierra del Fuego  2039           253948             128702   
743  Tierra del Fuego  2040           258020             130721   

     poblacion_mujeres  provincia_id  hogares  viviendas_particulares  \
0              1622915             2  1150134                 1423973   
1              1623804             2  1150134                 1423973   
2              1624755             2  1150134                 1423973   
3              1625737             2  1150134                 1423973   
4              1626722             2  1150134                 1423973   
..                 ...           ...      ...                     ...   
739             119026            94    38956                   43360   
740             121109            94    38956                   43360   
741             123183            94    38956                   43360   
742             125246            94    38956                   43360   
743             127299            94    38956                   43360   

     viviendas_particulares_habitadas  superficie_km2  densidad_poblacion  
0                             1082998             200        15142.405000  
1                             1082998             200        15168.195000  
2                             1082998             200        15194.300000  
3                             1082998             200        15220.380000  
4                             1082998             200        15246.145000  
..                                ...             ...                 ...  
739                             36689         1002445            0.241004  
740                             36689         1002445            0.245135  
741                             36689         1002445            0.249244  
742                             36689         1002445            0.253329  
743                             36689         1002445            0.257391  

[744 rows x 11 columns]

##### 3.8 - Comprobar si hay datos nulos

In [38]:
poblac_superficie.isnull().sum()

provincia                           0
anio                                0
poblacion_total                     0
poblacion_varones                   0
poblacion_mujeres                   0
provincia_id                        0
hogares                             0
viviendas_particulares              0
viviendas_particulares_habitadas    0
superficie_km2                      0
densidad_poblacion                  0
dtype: int64

##### 3.9 - Calculo del z-score de la densidad_población y agregar resultado en una nueva columna 'Fuera de rango'

In [39]:
poblac_superficie['fuera_de_rango'] = (poblac_superficie['densidad_poblacion'] - np.mean(poblac_superficie['densidad_poblacion'])) / np.std(poblac_superficie['densidad_poblacion'])
poblac_superficie

provincia  anio  poblacion_total  poblacion_varones  \
0     Capital Federal  2010          3028481            1405566   
1     Capital Federal  2011          3033639            1409835   
2     Capital Federal  2012          3038860            1414105   
3     Capital Federal  2013          3044076            1418339   
4     Capital Federal  2014          3049229            1422507   
..                ...   ...              ...                ...   
739  Tierra del Fuego  2036           241593             122567   
740  Tierra del Fuego  2037           245734             124625   
741  Tierra del Fuego  2038           249853             126670   
742  Tierra del Fuego  2039           253948             128702   
743  Tierra del Fuego  2040           258020             130721   

     poblacion_mujeres  provincia_id  hogares  viviendas_particulares  \
0              1622915             2  1150134                 1423973   
1              1623804             2  1150134                 1423973   
2              1624755             2  1150134                 1423973   
3              1625737             2  1150134                 1423973   
4              1626722             2  1150134                 1423973   
..                 ...           ...      ...                     ...   
739             119026            94    38956                   43360   
740             121109            94    38956                   43360   
741             123183            94    38956                   43360   
742             125246            94    38956                   43360   
743             127299            94    38956                   43360   

     viviendas_particulares_habitadas  superficie_km2  densidad_poblacion  \
0                             1082998             200        15142.405000   
1                             1082998             200        15168.195000   
2                             1082998             200        15194.300000   
3                             1082998             200        15220.380000   
4                             1082998             200        15246.145000   
..                                ...             ...                 ...   
739                             36689         1002445            0.241004   
740                             36689         1002445            0.245135   
741                             36689         1002445            0.249244   
742                             36689         1002445            0.253329   
743                             36689         1002445            0.257391   

     fuera_de_rango  
0          4.731966  
1          4.740389  
2          4.748916  
3          4.757434  
4          4.765849  
..              ...  
739       -0.213774  
740       -0.213773  
741       -0.213771  
742       -0.213770  
743       -0.213769  

[744 rows x 12 columns]

##### 3.10 - Calculo el percentil de la columna 'Fuera de rango'

In [41]:
p99 = np.percentile(poblac_superficie['fuera_de_rango'],99)
p99

np.float64(4.821306724666531)

In [43]:
poblac_superficie[poblac_superficie['fuera_de_rango'] >= p99]

provincia  anio  poblacion_total  poblacion_varones  \
13  Capital Federal  2023          3083770            1452588   
14  Capital Federal  2024          3085483            1454716   
15  Capital Federal  2025          3086680            1456560   
16  Capital Federal  2026          3087338            1458111   
17  Capital Federal  2027          3087434            1459359   
18  Capital Federal  2028          3086973            1460309   
19  Capital Federal  2029          3085971            1460970   
20  Capital Federal  2030          3084450            1461355   

    poblacion_mujeres  provincia_id  hogares  viviendas_particulares  \
13            1631182             2  1150134                 1423973   
14            1630767             2  1150134                 1423973   
15            1630120             2  1150134                 1423973   
16            1629227             2  1150134                 1423973   
17            1628075             2  1150134                 1423973   
18            1626664             2  1150134                 1423973   
19            1625001             2  1150134                 1423973   
20            1623095             2  1150134                 1423973   

    viviendas_particulares_habitadas  superficie_km2  densidad_poblacion  \
13                           1082998             200           15418.850   
14                           1082998             200           15427.415   
15                           1082998             200           15433.400   
16                           1082998             200           15436.690   
17                           1082998             200           15437.170   
18                           1082998             200           15434.865   
19                           1082998             200           15429.855   
20                           1082998             200           15422.250   

    fuera_de_rango  
13        4.822258  
14        4.825056  
15        4.827011  
16        4.828085  
17        4.828242  
18        4.827489  
19        4.825853  
20        4.823369

### 4 - Genera un informe breve sobre una de las variables originales (por ejemplo, superficie), con algunas medidas resumen (percentiles, promedio, etc.) y tus respuestas.

In [44]:
import numpy as np

# Seleccionamos la variable de interés
superficie = poblac_superficie["superficie_km2"]

# Medidas descriptivas
resumen = {
    "Promedio": superficie.mean(),
    "Mediana (P50)": np.percentile(superficie, 50),
    "Percentil 25 (Q1)": np.percentile(superficie, 25),
    "Percentil 75 (Q3)": np.percentile(superficie, 75),
    "Mínimo": superficie.min(),
    "Máximo": superficie.max(),
    "Desviación estándar": superficie.std()
}

print(resumen)

{'Promedio': np.float64(156719.75), 'Mediana (P50)': np.float64(101117.5), 'Percentil 25 (Q1)': np.float64(78272.75), 'Percentil 75 (Q3)': np.float64(157946.25), 'Mínimo': np.int64(200), 'Máximo': np.int64(1002445), 'Desviación estándar': np.float64(190131.34393628925)}


In [45]:
poblac_superficie["superficie_km2"].describe()

count    7.440000e+02
mean     1.567198e+05
std      1.901313e+05
min      2.000000e+02
25%      7.827275e+04
50%      1.011175e+05
75%      1.579462e+05
max      1.002445e+06
Name: superficie_km2, dtype: float64

## 📊 Informe de la variable superficie_km2

#### La variable superficie (km²) de las provincias analizadas muestra los siguientes resultados:

- Cantidad de observaciones: 744
- Promedio (media): 156,719 km²
- Mediana (P50): 101,117 km² → indica que la mitad de las provincias tienen menos de este valor.
- Rango intercuartílico (Q1–Q3): entre 78,273 km² y 157,946 km².
- Mínimo: 200 km²
- Máximo: 1,002,445 km²
- Desviación estándar: 190,131 km²